src/index.js

```diff
@@ -1,26 +1,29 @@
 import Vue from "vue";
 import VueRouter from "vue-router";
+import "animate.css";
 import App from "./App.vue";
 import router from "./helpers/router";
 import { store } from "./helpers/store";
 import * as fb from "./helpers/firebaseConfig";
 import Icon from "vue-awesome/components/Icon.vue";
+import VueMoment from "vue-moment";
 
 Vue.config.productionTip = false;
 Vue.config.devtools = process.env.NODE_ENV === "development";
 
+Vue.use(VueMoment);
 Vue.use(VueRouter);
 Vue.component("v-icon", Icon);
 
 new Vue({
   router,
   store,
   created() {
-    fb.auth.onAuthStateChanged((user) => {
+    fb.auth.onAuthStateChanged(async (user) => {
       console.log("User state changed", user);
-      this.$store.commit("setCurrentUser", user);
-      this.$store.dispatch("fetchUserProfile");
-      if (!user) {
+      await this.$store.commit("setCurrentUser", user);
+      if (!user && !this.$store.state.redirecting) {
+        console.warn("User not logged in, creating anonymous profile...");
```

Comment:
I'd really recommend you have some better logging infrastructure.

Consider using some logging lib that allows your to do stuff like `log.warn()`, `log.error()`, `log.info()`, `log.debug()` and `log.trace()` depending on the environment (e.g. `dev` vs `prod` will filter out different levels of logging).

Reply:
Definitely, I was wondering about this all along. I'll have a look after this one is merged